h/schemas/api/group.py

```diff
@@ -53,6 +54,7 @@ def validate(self, data):
 
         appstruct = super().validate(data)
         appstruct = self._whitelisted_fields_only(appstruct)
+        appstruct.setdefault("type", self.schema["properties"]["type"]["default"])
```

Comment:
If no `type` param is provided in the request then the schema injects `"private"` as the default value. This seems like a good approach because views and their tests can just assume that `appstruct["type"]` is always present and valid, they don't need to deal with it being potentially missing or injecting the default value.

This is different from how other optional fields are implemented in these groups APIs though: the schema just lets them be missing from the `appstruct` and the views have to deal with it. Should maybe consider doing `type` that way as well for consistency.